src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -181,14 +181,27 @@ await _dataImportService.UpdateStatus(dataImport.Id,
                             .Select(f => f.AutoSelectFilterItemLabel)
                             .SingleOrDefault() ?? "Total"; // If meta file didn't specify a default, look for "Total"
 
-                        return filterItems
+                        var filterItemsWithAutoSelectLabel = filterItems
                             .Where(item =>
-                                item.FilterGroup.FilterId == filterId
-                                && item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase))
-                            .Select(item => new { item.Id , item.Label})
-                            // There might be two filter items with the same label under different groups.
-                            // If so, we set no autoSelectFilterItem.
-                            .SingleOrDefault();
+                                item.FilterGroup.FilterId == filterId &&
+                                item.Label.Equals(autoSelectFilterItemLabel, OrdinalIgnoreCase));
+
+                        // There might be two or more filter items with the same label under different groups
```

Comment:
Would mention that the current implementation only considers a filter groups and will not work for filter hierarchies for the time-being.

We should create a follow-up bug / task ticket to address this correctly and update the comment with a TODO referencing this ticket number e.g. `TODO EES-blah: Fix auto-select filter items for filter hierarchies`